assertk/src/jvmMain/kotlin/assertk/failure.kt

```diff
@@ -18,3 +21,21 @@ internal actual inline fun Throwable.addSuppressed(error: Throwable) {
 }
 
 internal actual inline fun Throwable.isOutOfMemory(): Boolean = this is OutOfMemoryError
+
+internal object AssumptionFailure : Failure {
+    override fun fail(error: Throwable) {
+        failWithNotInStacktrace(
+            TestAbortedException(
```

Comment:
This will treat every exception as an abort, should we only catch `AssertionFailedError` exceptions so that other programming errors continue to fail the test?

Reply:
Added additional tests, It seems like https://github.com/willowtreeapps/assertk/pull/503 handles this